lib/controllers/ServiceFabrikAdminController.js

```diff
@@ -623,6 +601,75 @@ class ServiceFabrikAdminController extends BaseController {
         .send({}));
   }
 
+  startMaintenance(req, res) {
+    logger.info('req.body=>', req.body);
+    return maintenanceManager
+      .getMaintenaceInfo()
+      .then(body => {
+        if (body === null) {
+          return maintenanceManager
+            .startMaintenace(req.body, req.user)
+            .then(body => res
+              .status(201)
+              .send(body));
+        } else {
+          res
+            .status(403)
+            .send({
+              errorMessage: 'System already in maintenance mode',
+              maintenanceInfo: body
+            });
+        }
+      });
+  }
+
+  updateMaintenance(req, res) {
+    return maintenanceManager
+      .getMaintenaceInfo()
+      .then(body => {
+        if (body !== null) {
+          return maintenanceManager
+            .updateMaintenace(req.body.progress, req.body.state, req.user)
```

Comment:
The name req.body.progress ==> req.body.progressInfo/maintenanceInfo  would have been better. Or 
Instead of updateMaintenace(req.body.progress, req.body.state, req.user)
 --> updateMaintenace(req.body, req.user) would look cleaner if body doesn't contain other info which should not be passed to updateMaintenace() method.

Reply:
1. Naming is always subjective. Will leave it as-is. [Also please look at method signature of MaintenanceManager.updateMaintenance - this makes it clear that progress is actually progress info.]
2. All the params passed to this method are needed. [JSHint would have errored otherwise]

Reply:
"2. All the params passed to this method are needed. [JSHint would have errored otherwise]"
For this I was suggesting updateMaintenace(req.body, req.user) just like the startMaintenance() method.
Yes of course the method signature is correct in current code.